src/core/exec-invoke.c

```diff
@@ -73,6 +77,11 @@
 
 #define SNDBUF_SIZE (8*1024*1024)
 
+/* Project id range for disk quotas */
+#define PROJ_ID_MIN UINT32_C(2147483648)
+#define PROJ_ID_MAX UINT32_C(4294967294)
+#define PROJ_ID_CLAMP_INTO_QUOTA_RANGE(rnd) ((uint32_t) ((rnd) % (PROJ_ID_MAX - PROJ_ID_MIN + 1)) + PROJ_ID_MIN)
```

Comment:
Is this the right place to define these? chattr-util.h or introduce a dedicated fs-project.ch?